src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsVariables/Arguments.tsx

```diff
@@ -0,0 +1,193 @@
+
+    const newArgs = args.replaceAll('--', doubleDash).replaceAll('-', dash);
+    let stringArray = [];
+    let isQuoteOpen = false;
+
+    for (let i = 0; i < newArgs.length; i += 1) {
+      if (newArgs[i] === '"') {
+        isQuoteOpen = !isQuoteOpen;
+
+        stringArray.push(newArgs[i]);
+      } else if (newArgs[i] === ' ') {
+        if (!isQuoteOpen) {
+          stringArray.push(space);
+        } else {
+          stringArray.push(stringSpace);
+        }
+      } else {
+        stringArray.push(newArgs[i]);
+      }
+    }
+
+    form.setFieldValue(
+      'args',
+      stringArray
+        .join('')
+        .replaceAll(doubleDash, '--')
+        .replaceAll(dash, '-')
+        .replaceAll(stringSpace, ' ')
+        .replaceAll(space, '\n')
+    );
+    setPrettifiedState(true);
+  };
+
+  return (
+    <ConfigurationCard
+      title="Arguments"
+      description="Define arguments which will be passed to the test executor."
+      footerText={
+        <Text className="regular middle">
+          Learn more about{' '}
+          <a
+            href="https://kubeshop.github.io/testkube/executor-soapui/#using-parameters-and-arguments-in-your-tests"
+            target="_blank"
+          >
+            Arguments
+          </a>
+        </Text>
+      }
+      onConfirm={onClickSave}
+    >
+      <ArgumentsWrapper>
+        <CopyCommand command={argsValue} isBordered additionalPrefix="executor-binary" />
+        <Form
+          form={form}
+          name="general-settings-name-description"
+          onChange={onChange}
+          onFinish={onSaveForm}
+          initialValues={{args: argsValue}}
+        >
+          <Space size={16} direction="vertical" style={{width: '100%'}}>
+            <Text className="regular middle" color={Colors.slate400}>
+              Arguments passed to the executor (concatted and passed directly to the executor)
+            </Text>
+
+            <Form.Item name="args" style={{marginBottom: 0, marginTop: 0}}>
+              <div style={{position: 'relative'}}>
+                <Button
+                  onClick={prettifyArgs}
+                  disabled={isPrettified}
+                  style={{position: 'absolute', top: 12, right: 12, zIndex: 10}}
+                  $customType="transparent"
+                >
+                  Prettify
+                </Button>
+
+                <Input.TextArea
+                  autoSize={{
+                    minRows: 8,
+                  }}
+                  className="args-textarea"
+                  placeholder={`--flag="value value"
+--flag-123=value
+value
+`}
+                />
+              </div>
+            </Form.Item>
+          </Space>
+        </Form>
+      </ArgumentsWrapper>
+    </ConfigurationCard>
+  );
+};
+
+const ArgumentsWrapper = styled.div`
+  .args-textarea {
+    color: ${Colors.slate50};
+
+    font-family: ${Fonts.robotoMono};
+
+    resize: none;
+  }
+`;
+
+export default Arguments;
```

Comment:
overall Arguments components seems very big to me, I think we need to refactor, split it into small ones, to lower its complexity